OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -2226,6 +2231,42 @@ void testAutomaticJointReversal()
     ASSERT_EQUAL(accErr, 0.0, sqrt(integ_accuracy));
 }
 
+void testUserJointReversal()
+{
+    using namespace OpenSim;
+
+    cout << "\n==========================================================="
+         << "\n Test joint reversal set by user in old model file"
+         << "\n==========================================================="
+         << endl;
+
+    // Open model.
+    auto model = Model("double_pendulum_testReverse.osim");
+    //model.finalizeFromProperties();
+    model.initSystem();
+
+    // The topology is specified as follows:
+    //     [ground] <- (pin1) <- [rod1] <- (pin2) <- [rod2]
+    // but the "reverse" element of "pin2" is "true" so, after deserialization,
+    // the topology should be the following:
+    //     [ground] <- (pin1) <- [rod1] -> (pin2) -> [rod2]
```

Comment:
If I interpret this correctly the desired "user topology" has `rod1` as the child of two joints (`pin1` and `pin2`), which is not a valid tree, so the `MultibodyTreeMaker` will reverse the `pin2` with `rod2` as the true outboard but keep the sense of `pin2` coordinate as describing `rod1` w.r.t. `rod2`.  Is that what you intended?